gateleen-security/src/main/java/org/swisspush/gateleen/security/authorization/RoleMapperHolder.java

```diff
@@ -0,0 +1,56 @@
+package org.swisspush.gateleen.security.authorization;
+
+import java.util.regex.Pattern;
+
+/**
+ * Holds a pattern and implements hashcode/equals to use in collections.
+ *
+ * @author https://github.com/kusig [Markus Guenther]
+ */
+public class RoleMapperHolder {
+
+    /**
+     * The regex pattern to match for a role mapping
+     */
+    private Pattern pattern;
+    /**
+     * The resulting role if the given regex pattern matches
+     */
+    private String role;
+
+    /**
+     * Defines if the originalRole must be kept or not in the list of roles to apply against the ACL
+     */
+    private Boolean keepOriginal;
+
+    RoleMapperHolder(Pattern pattern, String role, Boolean keepOriginal) {
```

Comment:
Propose to add JavaDoc and explain that all three parameters must be !=null

Reply:
Could even shown with @NotNull in this case, any problem with that?